RELEASES.md

```diff
@@ -1,3 +1,148 @@
+Version 1.79.0 (2024-06-13)
+==========================
+
+<!-- FIXME: unsorted T-types relnotes -->
+- [check `FnDef` return type for WF](https://github.com/rust-lang/rust/pull/115538/)
+- [rework opaque type region inference](https://github.com/rust-lang/rust/pull/116891/)
+- [instantiate higher ranked goals outside of candidate selection](https://github.com/rust-lang/rust/pull/119820/)
+- [Pass list of defineable opaque types into canonical queries](https://github.com/rust-lang/rust/pull/122077/)
+- [Eagerly instantiate closure/coroutine-like bounds with placeholders to deal with binders correctly](https://github.com/rust-lang/rust/pull/122267/)
+- [Use fulfillment in method probe, not evaluation](https://github.com/rust-lang/rust/pull/122317/)
+- [Make inductive cycles always ambiguous](https://github.com/rust-lang/rust/pull/122791/)
+- [Enforce closure args + return type are WF](https://github.com/rust-lang/rust/pull/123531/)
+- [Fix trait solver overflow with `non_local_definitions` lint](https://github.com/rust-lang/rust/pull/123594/)
+- [More DefineOpaqueTypes::Yes](https://github.com/rust-lang/rust/pull/123794/)
+
+<a id="1.79.0-Language"></a>
+
+Language
+--------
+- [Stabilize inline `const {}` expressions.](https://github.com/rust-lang/rust/pull/104087/)
+- [Prevent opaque types being instantiated twice with different regions within the same function.](https://github.com/rust-lang/rust/pull/116935/)
+- [Stabilize WebAssembly target features that are in phase 4 and 5.](https://github.com/rust-lang/rust/pull/117457/)
+- [Add the `redundant_lifetimes` lint to detect lifetimes which are semantically redundant.](https://github.com/rust-lang/rust/pull/118391/)
+- [Stabilize the `unnameable_types` lint for public types that can't be named.](https://github.com/rust-lang/rust/pull/120144/)
+- [Enable debuginfo in macros, and stabilize `-C collapse-macro-debuginfo` and `#[collapse_debuginfo]`.](https://github.com/rust-lang/rust/pull/120845/)
+- [Propagate temporary lifetime extension into `if` and `match` expressions.](https://github.com/rust-lang/rust/pull/121346/)
+- [Restrict promotion of `const fn` calls.](https://github.com/rust-lang/rust/pull/121557/)
+- [Split `refining_impl_trait` lint into `_reachable`, `_internal` variants.](https://github.com/rust-lang/rust/pull/121720/)
```

Comment:
Specifically, the change as compared with stable Rust is that we're now linting refining impls of traits that are not reachable from outside the crate when we were not doing that before.

Perhaps most explicitly:

```suggestion
- [Rename the existing `refining_impl_trait` lint to `refining_impl_trait_reachable`; add `refining_impl_trait_internal` lint to warn against impls that refine a trait not reachable from outside the crate; place these both in a `refining_impl_trait` lint group.](https://github.com/rust-lang/rust/pull/121720/)
```

Or maybe more concisely:

```suggestion
- [Warn against refining impls of crate-private traits with `refining_impl_trait` lint.](https://github.com/rust-lang/rust/pull/121720/)
```